saleor/payment/gateways/adyen/plugin.py

```diff
@@ -242,16 +247,37 @@ def process_payment(
             kind = TransactionKind.CAPTURE
         else:
             kind = TransactionKind.AUTH
+
+        action = result.message.get("action")
+        error_message = result.message.get("refusalReason")
+        if action:
+            _type, payment_id = from_global_id(payment_information.payment_id)
+            try:
+                payment = Payment.objects.get(pk=payment_id)
+                details = result.message.get("details")
+                payment.extra_data = json.dumps(
+                    {
+                        "payment_data": action["paymentData"],
+                        "parameters": [detail["key"] for detail in details]
+                        if details
```

Comment:
This `if else` will not be needed if you assign `list` in `get` call:
`details = result.message.get("details", [])`